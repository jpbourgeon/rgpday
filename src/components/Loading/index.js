import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import CircularProgress from '@material-ui/core/CircularProgress'

const styles = theme => ({
  layout: {
    width: 'auto',
    height: '100vh'
  },
  container: {
    height: '100%'
  },
  item: {
    textAlign: 'center'
  }
})

function Loading (props) {
  const { classes } = props

  return (
    <div className={classes.layout}>
      <Grid container alignContent='center' alignItems='center' className={classes.container}>
        <Grid item xs={12} className={classes.item}>
          <CircularProgress color='secondary' />
        </Grid>
      </Grid>
    </div>
  )
}

Loading.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(Loading)
