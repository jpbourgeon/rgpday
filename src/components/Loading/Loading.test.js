/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import Loading from './'
import jssSerializer from '../../../utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The Loading component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Loading />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
