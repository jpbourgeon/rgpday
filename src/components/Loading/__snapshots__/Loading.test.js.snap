// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Loading component should render correctly 1`] = `null`;
